### PR TITLE
Codemod for auto-creating `@jest/globals` imports

### PR DESCRIPTION
**Pull request by danbeam**
I recently had to convert a bit of code that used jest globals (e.g. `describe()`, `it()`, `expect()`) to explicitly importing these APIs from `@jest/globals`.

Before:
```ts
it('works', () => {
  expect(true).toBe(true);
});
```

After:
```ts
import { expect, it } from '@jest/globals';

it('works', () => {
  expect(true).toBe(true);
});
```

While doing that manually I thought: this would make a good codemod!

So I wrote a fairly simple one that actually scrapes the exports from the `@jest/globals` package itself[1], finds global usage, and adds imports.

[1] I don't use `require.resolve()` nor actually `import` it cuz both are enforcedly disallowed, which is a mild bummer, but overall I think it's still kinda slick.  If you know a better way, let me know!

### CODE REVIEW

**Comment by skovhus** [comment]
Interesting! Thanks for another contribution! I just read quickly through this, and here is my feedback.

Note that we have similar code for this for all the transformations – in the CLI you can pick what we internal call `standaloneMode` if this you say no to "Will you be using Jest on Node.js as your test runner?". This is done here: https://github.com/skovhus/jest-codemods/blob/main/src/utils/finale.ts#L25 Not sure how many people use this feature... It seems we might want to use `@jest/globals` instead of `jest-mock` and `expect` as we do right now. 

We also have a bunch of import helpers defined here, that could be useful for this PR. Maybe they handle more corner cases than you do, or maybe you handle additional corner cases: https://github.com/skovhus/jest-codemods/blob/main/src/utils/imports.ts

Currently all codemods converts from one test runner to another, this one is different. I guess it shouldn't be exposed in the CLI or? But the transformation here could just be exposed directly as these: https://github.com/skovhus/jest-codemods#usage-jscodeshift

**Comment by danbeam** [comment]
@skovhus sorry for the lag on this; do you want the content of this PR (a `@jest/globals` codemod)?  It wasn't clear from your previous response.

If "yes", I can probably find some time to e.g. update to `utils/imports.ts` or something along those lines.  But I kinda wanted to know whether you actually want the content first 😅 .

If "no": I can close.

> Currently all codemods converts from one test runner to another, this one is different. I guess it shouldn't be exposed in the CLI or? But the transformation here could just be exposed directly as these: https://github.com/skovhus/jest-codemods#usage-jscodeshift

I don't have strong feelings here; whatever you prefer.

**Comment by skovhus** [comment]
This is great.  Let us just document it in the README and then I think we are good to go.

**Comment by danbeam** [comment]
> This is great. Let us just document it in the README and then I think we are good to go.

I can add more prose about this codemod, but it seems fairly discretionary about what you'd like to say (e.g. do you want another sentence before the status images?  I don't know that this is worth highlighting like that.)

File: src/transformers/jest-globals-import.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import fs from 'fs'
+import type { ImportSpecifier, JSCodeshift } from 'jscodeshift'
+import path from 'path'
+
+import { findImports, removeRequireAndImport } from '../utils/imports'
+
+const jestGlobals = new Set<string>()
+
+const jestGlobalsPath = path.join(
```
Comment (skovhus) [inline src/transformers/jest-globals-import.ts:9]: This import looks a bit funky. Are we importing a type definition file? I'm also pretty sure this will not work in various package system as we assume where dependencies are placed. We should import this like any other third parties.

File: package.json [hunk 2]
```diff
@@ -63,6 +63,7 @@
     "update-notifier": "5.1.0"
   },
   "devDependencies": {
+    "@jest/globals": "29.3.1",
```
Comment (skovhus) [inline package.json:66]: It seems like we are using this as a production dependency now (production is everything that is required to run the codemod, devDependencies is everything required to develop jest-codemods).

Also we need to update the lock file.

File: src/transformers/jest-globals-import.ts (hunk 1 above)
Comment (danbeam) [reply src/transformers/jest-globals-import.ts:9]: this is definitely the funkiest part.  yes, this is reading a type file.  the point is to dynamically parse the exported members.  yes, it is possibly risky.

we can freeze the list of things exported from jest, it'll just be a little behind / not update automatically.  that could be the right set of tradeoffs for this library (rather than a more homogenous environment).

do you want me to do that?  I think there's already a list of jest-y things in `jest-codemods`, right?

File: package.json (hunk 2 above)
Comment (danbeam) [reply package.json:66]: > It seems like we are using this as a production dependency now (production is everything that is required to run the codemod, devDependencies is everything required to develop jest-codemods).

Ah; fair enough.  I think whether this is a dev dependency, prod dependency, or dependency at all is contingent on the outcome of whether we parse files (i.e. https://github.com/skovhus/jest-codemods/pull/508#discussion_r1401088600)

> 
> Also we need to update the lock file.

I had tried this (running `pnpm install`) but I wasn't seeing any difference in the lock file.  I believe it's because the exact same version is already there(?).  I don't use pnpm much so let me know if I'm messing something up!

File: src/transformers/jest-globals-import.ts (hunk 1 above)
Comment (skovhus) [reply src/transformers/jest-globals-import.ts:9]: Hardcoding them here should be good enough. 

Bonus point for adding a unit test that verifies that the hardcoded list is up to date with Jest.

**Comment by danbeam** [comment]
@skovhus anything else you need me to do here?  I haven't figured out how to get your CI checks passing.  Something about a lock file.

**Comment by danbeam** [comment]
hey @skovhus 👋 

just following up again here: anything I can do to get this done?  it's not particularly urgent, I'm just hoping to tie up loose ends by getting this in eventually!  (when it's right by you)

let me know if you can!  (if you're on holiday or something no sweat.)

**Comment by skovhus** [comment]
Thanks for adding support for this!
